### PR TITLE
Reactivate highlighting feature for log entries

### PR DESCRIPTION
**Pull request by AndiPersti**
This PR addresses #1873 

We need to load the necessary JS files when at least one link contribution is shown in the logs.

I've also removed an obsolete version of the jQuery.scrollTo library.

### CODE REVIEW

File: src/Template/Element/logs/log_entry_annexe.ctp [hunk 1]
```diff
@@ -49,10 +49,12 @@ $langDir = LanguagesLib::getLanguageDirection($langCode);
             } elseif ($type == 'license') {
                 echo $sentenceLink.' ➜ '.$this->Html->tag('span', $sentenceText, array('class' => 'license'));
             } else { // link
+                $this->Html->script(['jquery.scrollTo.min.js', 'sentences.logs.js'],
+                                    ['block' => 'scriptBottom']);
```
Comment (trang) [inline src/Template/Element/logs/log_entry_annexe.ctp]: One minor thing: if the scripts are included from here, then they will be loaded in `/users/show` where they are not needed. They are only needed in `/sentences/show` so it is better to include them from there.

Comment (AndiPersti) [reply src/Template/Element/logs/log_entry_annexe.ctp]: Yes, you're right.

Another thing I haven't considered in my current implementation is, that it will load the scripts if a sentence currently doesn't have any translation but the logs show linking and corresponding unlinking actions, i.e. it had a translation in the past which was deleted later.

I don't know if there are any cases like that in the production database but I'm thinking about avoiding loading the scripts in that case.

Comment (trang) [reply src/Template/Element/logs/log_entry_annexe.ctp]: There are many cases of sentences that don't have any translations and indeed, the script would be unnecessary for those too.

Comment (AndiPersti) [reply src/Template/Element/logs/log_entry_annexe.ctp]: I was more thinking of a case like this:
example_logs

I've just pushed my new implementation.

File: src/Template/Sentences/show.ctp [hunk 2]
```diff
@@ -150,14 +148,25 @@ echo $this->element('/sentences/navigation', [
...
+                }
+                if ($contribution->type == "link" and $contribution->action == "delete") {
+                    unset($active_links[$contribution->translation_id]);
+                }
             }
             echo '</md-list>';
+            if (!empty($active_links)) {
+                $this->Html->script(['jquery.scrollTo.min.js', 'sentences.logs.js'],
+                                    ['block' => 'scriptBottom']);
+            }
```
Comment (trang) [inline src/Template/Sentences/show.ctp]: A more simple way would be to just check if `$sentence->translations` is empty. Analyzing the logs works just fine too, but it's a bit more lines of code :)

Now that doesn't take into account that logs are not 100% reliable (#1653). And if you would want to be completely accurate about when to load the javascript, you could keep your code above and additionally check if the translations in the `$active_links` actually exist in `$sentence->translations`.

But if you ask me, we don't have to go into that level of details. Such cases would be rare and we don't have to over-complexify the code for a low impact optimization.

Comment (AndiPersti) [reply src/Template/Sentences/show.ctp]: Yeah, after sleeping over it, I realized that I was overthinking the problem. :-)

But while testing for a sentence without a log (by manually inserting it into the sentences table using the mysql REPL) I've noticed that the condition on line 152 doesn't work: https://github.com/Tatoeba/tatoeba2/blob/a171662ded228c0e5be38a56b6504ea009f68e38/src/Template/Sentences/show.ctp#L152

I've expected to get the message "There is no log for this sentence" from the else branch. But since `$contributions` is a CakePHP ResultSet it will never be empty because an empty ResultSet is an object containing an empty list. Thus I think a better way for testing whether `$contributions` is empty would be to check its count.

I've updated the PR (hopefully the last time :-) )

File: src/Template/Sentences/show.ctp [hunk 3]
```diff
@@ -149,7 +147,11 @@ echo $this->element('/sentences/navigation', [
             echo $this->Sentences->originText($sentence);
         }
 
-        if (!empty($contributions)) {
+        if ($contributions->count() > 0) {
+            if (!empty($sentence->translations)) {
```
Comment (trang) [inline src/Template/Sentences/show.ctp]: You're probably going to hate me but I realized while testing that `$sentence->translations` will (should?) never be empty. It's structured into two arrays.
```
[
	(int) 0 => [], // direct translations
	(int) 1 => [] // indirect translations
]
```
So we need to test that `$sentence->translations[0]` is empty.

Comment (AndiPersti) [reply src/Template/Sentences/show.ctp]: No problem, I can tolerate a lot of criticism (especially if it's justified) :-)

You probably won't believe me but I originally wrote the test with `$sentence->translations[0]` but just before committing I thought for some reason it's not necessary to be that specific and then forgot to also test for sentences without any translations where that makes a difference.
